samza-core/src/main/java/org/apache/samza/task/RunLoop.java

```diff
@@ -457,17 +457,25 @@ private void process() {
       final IncomingMessageEnvelope envelope = state.fetchEnvelope();
       log.trace("Process ssp {} offset {}", envelope.getSystemStreamPartition(), envelope.getOffset());
 
-      final ReadableCoordinator coordinator = new ReadableCoordinator(task.taskName());
-      TaskCallbackFactory callbackFactory = new TaskCallbackFactory() {
-        @Override
-        public TaskCallback createCallback() {
-          state.startProcess();
-          containerMetrics.processes().inc();
-          return callbackManager.createCallback(task.taskName(), envelope, coordinator);
-        }
+      Runnable processWorker = () -> {
+        final ReadableCoordinator coordinator = new ReadableCoordinator(task.taskName());
+        TaskCallbackFactory callbackFactory = new TaskCallbackFactory() {
+          @Override
+          public TaskCallback createCallback() {
+            state.startProcess();
+            containerMetrics.processes().inc();
+            return callbackManager.createCallback(task.taskName(), envelope, coordinator);
+          }
```

Comment:
This is not related to your change. Currently the life-cycle of the executor service is controlled two levels up at the `SamzaContainer` layer, i.e, the executor service is stopped as a part of the `SamzaContainer` shutdown sequence. 

This makes reasoning about life-cycle of the executor service harder. It would be better to manage the life-cycle entirely within the RunLoop and stop it as a part of `RunLoop` shutdown sequence.

What do you think?

Reply:
Not applicable anymore.